.github/workflows/install-and-deploy.yml

```diff
@@ -0,0 +1,55 @@
+name: install-and-deploy
+
+on:
+  workflow_call:
+    inputs:
+      node-version:
+        description: "Node version"
+        required: true
+        type: string
+      publish:
+        description: "Publish"
+        required: true
+        type: boolean
+      cache:
+        description: "Use cache node modules"
+        required: false
+        type: boolean
+
+jobs:
+  install:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.node-version }}
+          registry-url: https://registry.npmjs.org/
+      - name: cache node modules
+        if: ${{ inputs.cache == true }}
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: install dependencies
+        run: npm install
+
+      - name: validate
+        run: npm run validate.ci
+
+      - name: publish
+        if: ${{ inputs.publish == true }}
+        run: npm publish --access=public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
```

Comment:
Changes to the workflows are OK.

The error in the workflow run is expected since the new workflow will not existing in the main branch until the PR I is merged.